catalog/plugin/catalog-plugin-async/src/main/java/org/codice/ddf/catalog/plugin/async/api/internal/ProcessingFramework.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.async.api.internal;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ *
+ * The {@code ProcessingFramework} processes all requests submitted to it. Available requests for processing
```

Comment:
I don't think I agree. It's a good practice to separate API and implementation from the get-go, even for experimental code.

It's fine if experimental APIs live in separate bundles, but please don't bundle the API and impl together. That quickly leads to a dependency nightmare as clients become unable to distinguish what they actually depend on, and as a result it becomes too easy to couple to implementation details instead of interfaces.